File: java/program87.java
import java.util.*;

class program87
{
    public static boolean CheckDigit(int No,int search)
    {   
        int iDigit = 0;

        while(No != 0)
        {
            iDigit = No % 10;
            if(iDigit == search)
            {
                break;
            }
            No = No / 10;
        }
        if(No == 0)
        {
            return false;
        }
        else
        {
            return true;
        }
    }

    public static void main(String arg[])
    {
        Scanner sobj = new Scanner(System.in);
        boolean bRet = false;

        System.out.println("Enter number : ");
        int value = sobj.nextInt();

        System.out.println("Enter the digit you want to search : ");
        int digit = sobj.nextInt();

        bRet = CheckDigit(value,digit);
        if(bRet == true)
        {
            System.out.println(digit+" is present...");
        }
        else
        {
            System.out.println(digit+" is not present ...");
        }
    }
}

File: cpp/program87.cpp
//program59.c

#include<iostream>
using namespace std;

class Digit
{
    public:
        int iNo;

        Digit(int X)
        {
            iNo = X;
        }

        bool CheckDigit(int iSearch)
        {
            int iDigit = 0;
            int temp = iNo;

            while(temp != 0)
            {
                iDigit = temp % 10;
                if(iDigit == iSearch)
                {
                    break;
                }
                temp = temp / 10;
            }
            if(iDigit == iSearch)
            {
                return true;
            }
            else
            {
                return false;
            }
        }
};

int main()
{
    int iValue = 0;
    int iNum = 0;
    bool bRet = false;

    cout<<"Enter number \n";
    cin>>iValue;

    cout<<"Enter digit between (0 to 9) to serch \n";
    cin>>iNum;

    Digit obj(iValue);

    bRet = obj.CheckDigit(iNum);
    if(bRet == true)
    {
        cout<<iNum<<" is present\n";
    }
    else
    {
        cout<<iNum<<" is NOT present \n";
    }

    return 0;
}